src/reporter/tests/test_NTNE.py

```diff
@@ -0,0 +1,197 @@
+        "index": [
+            "1970-01-30T00:00:00.000"
+        ],
+        "type": 'Kitchen'
+    },
+    {
+        "id": 'Room0',
+        "index": [
+            "1970-01-30T00:00:00.000"
+        ],
+        "type": 'Room'
+    }]
+
+    expected = exp_values
+
+    assert obtained == expected
+
+def test_not_found():
+    r = requests.get(query_url())
+    assert r.status_code == 404, r.text
+    assert r.json() == {
+        "error": "Not Found",
+        "description": "No records were found for such query."
+    }
+
+def test_NTNE_type(reporter_dataset):
+    # Query
+    query_params = {
+        'type': entity_type
+    }
+    r = requests.get(query_url(), params=query_params)
+    assert r.status_code == 200, r.text
+
+    # Assert
+    obtained = r.json()
+    expected_type = 'Room'
+    expected_values = list(range(n_days))
+    expected_index = [
+        '1970-01-{:02}T00:00:00'.format(i+1) for i in expected_values
+    ]
+    expected = [{
+        'id': 'Room0',
+        'index': [expected_index[-1]+'.000'],
+        'type': expected_type
+    }]
+    assert obtained == expected
+
+def test_NTNE_fromDate_toDate(reporter_dataset):
+    # Query
+    query_params = {
+        'fromDate': "1970-01-06T00:00:00",
+        'toDate': "1970-01-17T00:00:00",
+    }
+    r = requests.get(query_url(), params=query_params)
+    assert r.status_code == 200, r.text
+
+    # Expect only last N
+    expected_values = list(range(5, 17))
+    expected_type = 'Room'
+    expected_id = 'Room0'
+    expected_type_1 = 'Kitchen'
+    expected_id_1 = 'Kitchen0'
+    expected_index = [
+        '1970-01-{:02}T00:00:00.000'.format(i+1) for i in expected_values
+    ]
+    assert len(expected_index) == 12
+    assert expected_index[0] == "1970-01-06T00:00:00.000"
+    assert expected_index[-1] == "1970-01-17T00:00:00.000"
+
+    # Assert
+    obtained = r.json()
+    expected = [{
+        'id': expected_id_1,
+        'index': [expected_index[-1]],
+        'type': expected_type_1
+    },
+    {
+        'id': expected_id,
+        'index': [expected_index[-1]],
+        'type': expected_type
+    }]
+    assert obtained == expected
+
+def test_NTNE_limit(reporter_dataset):
+    # Query
+    query_params = {
+        'limit': 1
+    }
+    r = requests.get(query_url(), params=query_params)
+    assert r.status_code == 200, r.text
+
+    expected_type = 'Room'
+    expected_id = 'Room0'
+    expected_index = [
+        '1970-01-30T00:00:00.000'
+    ]
+
+    # Assert
+    obtained = r.json()
+    expected = [{
+        'id': expected_id,
+        'index': expected_index,
+        'type': expected_type
+    }]
+    assert obtained == expected
+
+def test_NTNE_offset(reporter_dataset):
+    # Query
+    query_params = {
+        'offset': 1
+    }
+    r = requests.get(query_url(), params=query_params)
+    assert r.status_code == 200, r.text
+
+    # Expect only last N
+    expected_values = list(range(1, n_days))
+    expected_type = 'Kitchen'
```

Comment:
https://travis-ci.org/smartsdk/ngsi-timeseries-api/builds/620582396#L1289 shows that you may get room actually. to be checked.

Reply:
Output is quite unpredictable for me. Sometime it shows Room and the sometime Kitchen. Don't know how to resolve this. Although all the test cases are passed now.

Reply:
> Output is quite unpredictable

I bet it's cos the result set isn't ordered by time. So you'll get the results in a different order every time you run the test. Can you check this? Thank you so much!

Reply:
I found the cause of failed testcases that was because of the time of data inserted. Both Room and Kitchen value is inserted at the same time. To resolve this, I have updated the test cases by changing the value of time-index.

Reply:
good catch!